### PR TITLE
Reset email settings

### PR DESCRIPTION
**Pull request by VincentMoyo**
Created reset email function
Added activity loaders 
Added more unit testing
Added new profile picture navigation button

### CODE REVIEW

File: CoolCurrency/Controller/SettingsViewController.swift [hunk 1]
```diff
@@ -196,6 +200,27 @@ extension SettingsViewController {
         present(alert, animated: true, completion: nil)
     }
     
+    private func resetEmailAlert() {
+        var textField = UITextField()
+        let alert = UIAlertController(title: "Set Email",
```
Comment (bgouws) [inline CoolCurrency/Controller/SettingsViewController.swift]: You can also put this in an extension to make things a little more streamlined. Junaid's last PR has a good example if you want to check it out.

**Review by bgouws — CHANGES_REQUESTED** [review]
Just a small  enhancement 😄

File: CoolCurrency/CurrencyViewModel/NavigationViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    }
+    
+    func loadUserSettingsFromDatabase() {
+        databaseRepository.retrieveUserInformationFromDatabase(userID: authenticationRepository.signedInUserIdentification(), completion: { [weak self] result in
+            do {
+                let newUserDetails = try result.get()
+                self?.userSettingsList = newUserDetails
+                self?.checkUserList()
+                self?.downloadProfileImageFromDatabase()
+                // self?.delegate?.bindViewModel()
```
Comment (bgouws) [inline CoolCurrency/CurrencyViewModel/NavigationViewModel.swift]: Some commented code here?

File: CoolCurrency/CurrencyViewModel/SettingsViewModel.swift [hunk 3]
```diff
@@ -37,7 +37,7 @@ class SettingsViewModel: SettingsViewModiable {
                 self?.userSettingsList = newUserDetails
                 self?.checkUserList()
                 self?.downloadProfileImageFromDatabase()
-                self?.delegate?.bindViewModel()
+                 self?.delegate?.bindViewModel()
```
Comment (bgouws) [inline CoolCurrency/CurrencyViewModel/SettingsViewModel.swift]: Think there is an extra space here?

File: CoolCurrency/UserAlerts.swift [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+                                      preferredStyle: .alert)
+        
+        alert.addAction(UIAlertAction(title: "OK",
+                                      style: .default,
+                                      handler: action))
+        
+        viewController.present(alert, animated: true, completion: nil)
+    }
+    
+    static func showUpdateFirstNameAlert(for viewController: UIViewController,
```
Comment (bgouws) [inline CoolCurrency/UserAlerts.swift:20]: So this are still a little bit too hard-coded, if I want to show a different alert I sort of need to come and add another function here, rather see if you can make one function that can handle these scenarios even if you have to pass through a few parameters 😄

**Review by bgouws — CHANGES_REQUESTED** [review]
Just some clean up and enhancements 😄

File: CoolCurrency/Controller/CryptoAndMetalViewController.swift [hunk 5]
```diff
@@ -29,15 +30,21 @@ class CryptoAndMetalViewController: UIViewController {
     
     override func viewDidLoad() {
         super.viewDidLoad()
+        activateActivityIndicatorView()
         currencyPicker.delegate = self
         currencyPicker.dataSource = self
-        currencyPicker.setValue(UIColor.white, forKeyPath: "textColor")
+        currencyPicker.setValue(UIColor(named: "SecondaryNavy"), forKeyPath: "textColor")
```
Comment (bgouws) [inline CoolCurrency/Controller/CryptoAndMetalViewController.swift]: You can add this to your style kit 😄 also consider the naming convention I have used below:

```
static var secondaryColour: UIColor {
   return UIColor(named: "secondaryColour")
}

File: CoolCurrency/Controller/HomeCurrencyViewController.swift [hunk 6]
```diff
@@ -23,10 +24,11 @@ class HomeCurrencyViewController: UIViewController {
     
     override func viewDidLoad() {
         super.viewDidLoad()
+        activateActivityIndicatorView()
+        currencyPickerView.setValue(UIColor(named: "SecondaryNavy"), forKeyPath: "textColor")
```
Comment (bgouws) [inline CoolCurrency/Controller/HomeCurrencyViewController.swift]: Style kit

**Review by bgouws — CHANGES_REQUESTED** [review]

**Review by bgouws — APPROVED** [review]

**Review by AbdullahNana10 — APPROVED** [review]

**Pull request merged**